### ISSUE TITLE
The loss of generator

### ISSUE THREAD

**Issue by pengliu380** [aelnouby/Text-to-Image-Synthesis#3]
Hello, author. I want to reproduce the experimental results. When I trained the model, I found that the loss of generator was always high and it looked like the generator was much weaker than the discriminator, what should I do?

**Comment by aelnouby** [comment]
I am working now on better ways to make the training more stable including implementing WGAN, minibatch discrimination and feature matching. I will mention you in the pull request to let you know.

Meanwhile I guess you can to train the generator more than the discriminator if you think it is weaker, this trick is used heavily in WGAN but the discriminator is the one that is trained more because it can be overpowered easily.

**Comment by aelnouby** [comment]
@pengliu380 I have added WGAN implementation to master, however I am still working on making gradient penalty work but it seems there is a bug in pytorch right now, I have submitted an issue.

Nevertheless, the WGAN implementation should be working but it is quite slow, attached an example of the Wasserstien estimate plot. 

![newplot](https://user-images.githubusercontent.com/8495451/33102329-42a7f460-ceea-11e7-9f37-4efb28d0f919.png)
